### ISSUE TITLE
Implement removal of layers

### ISSUE THREAD

**Issue by astrofrog** [WorldWideTelescope/pywwt#125]
We should implement a ``layer.remove()`` method that removes layers from the visualization.

**Comment by philrosenfield** [comment]
+1 APLpy has this, and I’ve found it awkward (last time I’ve used it) to remove a layer with multiple plotted. Especially for notebook and lab users, if the layer isn’t saved as a variable, I couldn’t remove it. 

I’m sorry to not have had time to dig into the code more deeply before suggesting this, but do layers get added to a list that can always be accessed?

**Comment by astrofrog** [comment]
@philrosenfield - not yet, see https://github.com/WorldWideTelescope/pywwt/issues/132 :)

**Comment by astrofrog** [comment]
It looks like layer removal doesn't work at the moment with the JS API:

```
wwtlib.LayerManager.deleteLayerByID(layer.id)
```

doesn't seem to do anything - it removes the layer from the manager, but the layer is still visible. I also tried ``layer.cleanUp`` but no luck.

**Comment by astrojonathan** [comment]
The functionality is already there, you just have to use all three parameters, if you skip the last two parameters it defaults as false and does nothing. Since the WWT layer universe is a multiverse (one for ambient/editing tours, one for playing tours) the delete can have different effects on the current state.

DeleteLayerByID(Guid ID, bool removeFromParent, bool updateTree)

Remove from parent will remove a tour from being referenced.. Must be true or the parental reference will still keep it alive.

Update tree will cause the UI to refresh to show the changes. 

In short when you use JavaScript to access c# projected methods, make sure you are not short-changing the parameter list. JavaScript is duck-typed and will let you call methods with too few too many or wrongly type parameters and hilarity ensues, so make sure you are checking the parameter list.

**Comment by astrofrog** [comment]
Just confirming that fixes it - thanks!
